### ISSUE TITLE
Recent 2.1.5 release is not compatible with Java 8

### ISSUE THREAD

**Issue by vlsi** [webcompere/system-stubs#79]
Could you please support Java 8 for a while?

We use system-stubs in pgjdbc, and we get CI failure if upgrading to 2.1.5

https://github.com/pgjdbc/pgjdbc/actions/runs/6989871071/job/19018844344#step:8:281

```
  /home/runner/work/pgjdbc/pgjdbc/pgjdbc/src/test/java/org/postgresql/jdbcurlresolver/PgServiceConfParserTest.java:17: error: cannot access uk.org.webcompere.systemstubs.environment.EnvironmentVariables
  import uk.org.webcompere.systemstubs.environment.EnvironmentVariables;
                                                  ^
    bad class file: /home/runner/.gradle/caches/modules-2/files-2.1/uk.org.webcompere/system-stubs-core/2.1.5/bd5eec4e841623f82e39720275e96ae5c6c925d5/system-stubs-core-2.1.5.jar(uk/org/webcompere/systemstubs/environment/EnvironmentVariables.class)
      class file has wrong version 55.0, should be 52.0
      Please remove or make sure it appears in the correct subdirectory of the classpath.
```

I suggest building with Java 21 (or 17) and use `--release 8` so you get bytecode runnable with Java 8.

**Comment by ashleyfrieze** [comment]
You don't need the version 2 release train. You can continue using version 1 with Java 8

**Comment by vlsi** [comment]
Can you clarify? Version 2.0.2 works with Java8, and it was unexpected (judging by version number) that 2.1.5 dropped Java 8 compatibility

**Comment by ashleyfrieze** [comment]
The reason for the version 2 release train was to provide an alternative to the reflection based manipulation of environment variables. This was required for Java 16 onwards.

The Readme clearly states that for  lower Java versions, the original v1 releases are sufficient.

If you're arguing about the use of semver, the change to 2.1 from 2.0 indicates a breaking change.

Stick with a lower version number than 2.1.x and things will be fine.

Java 8 is 9 years old and already end of life. Java 11 is 5 years old and also end of life. Upgrade your project. System Stubs works from Java 11 onwards.

**Comment by vlsi** [comment]
I mean I could understand if you dropped Java 8 in 2.x or 3.x release. It just looks surprising to drop Java 8 with 2.0-2.1 version change.

Java 8 will be around at least till 2030.

**Comment by ashleyfrieze** [comment]
@vlsi in version 2.0.0 the README warned that Java 8 support was not guaranteed for the future - https://github.com/webcompere/system-stubs/blob/system-stubs-parent-2.0.0/README.md?plain=1#L20

Given you wish to stay on Java 8 long past its useful life, my recommendation is to move back to the `1.2.0` version of System Stubs. There's no reason to have ever upgraded to the v2 release train, and there are some concerns with v2.0.x which we fixed in v2.1.x. These relate to the use of mockito, and multi-threaded tests that depend on environment variables.

v1.x System Stubs is compatible with Java 8 - 15
v2.0.x System Stubs is compatible with Java 8+
v2.1.x System Stubs is compatible with Java 11+

I would heavily recommend avoiding using the same Java edition any longer than the time it takes to get access to the next LTS version.

**Comment by ashleyfrieze** [comment]
Compatibility matrix explained.
